src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -84,14 +193,62 @@ export default class FileField extends React.Component {
   };
 
   removeFile = index => {
-    const newFileList = this.props.formData.filter(
-      (file, idx) => index !== idx,
-    );
+    const newFileList = this.props.formData.filter((__, idx) => index !== idx);
     if (!newFileList.length) {
       this.props.onChange();
     } else {
       this.props.onChange(newFileList);
     }
+    this.setState({
+      files: this.state.files.filter(entry => entry.index !== index),
+    });
+    // clear file input value; just in case
+    if (this.fileInputRef.current) {
+      this.fileInputRef.current.value = '';
+    }
+    this.focusAddAnotherButton();
+  };
+
+  /**
+   * @typeof Files~ApiResponse
+   * @type {object}
+   * @property {string} name - file name
+   * @property {string} confirmationCode - uuid of uploaded file
+   * @property {string} attachmentId - form ID set by user
+   */
+  /**
+   * @typedef Files~render
+   * @type {object}
+   * @property {Files~ApiResponse} file - File object, used when user submits
+   *   password
+   * @property {string} name - file name for UI
+   * @property {boolean} isEncrypted - encrypted state
+   * @property {boolean} isTemporary - temporary file listing added
+   *   before uploading to have a placeholder in the UI
+   */
+  /**
+   * Get file list. it needs to add files that have been marked as encrypted in
+   * the state file data
+   * @return {Files~render}
+   */
+  getFiles = () => {
+    // remove temp placeholders; just in case
+    const fileList =
+      this.props.formData?.filter(file => !file.isTemporary) || [];
+    this.state.files.forEach(({ file, index, isEncrypted }) => {
+      if (file.name === fileList[index]?.name) {
+        // already uploaded; remove from state list
+        return false;
+      }
+      fileList.push({
+        file,
+        name: file.name,
```

Comment:
Is this necessary?

Reply:
So the `formData` saves the server response

```
  /**
   * @typeof Files~ApiResponse
   * @type {object}
   * @property {string} name - file name
   * @property {string} confirmationCode - uuid of uploaded file
   * @property {string} attachmentId - form ID set by user
   */
```

where as the `this.state.files` stores

```
  /**
   * @typedef Files~state - File state object
   * @type {object}
   * @property {object|null} file - upload API supplied File object: see
   *   https://developer.mozilla.org/en-US/docs/Web/API/File
   * @property {number} index - index matching formData array position
   * @property {boolean} isEncrypted - detected encrypted state of a PDF file
   * @property {boolean} isTemporary - indicates state data that needs needs a
   *   placeholder
   */
```

the `getFiles` function needs to add the state object into the form data files for rendering - the `state.files` haven't been uploaded yet,  so they won't be in the form  data. Including a `name` makes it so we don't have to check if there is a difference in the file list that is to be rendered. Also the `getFiles` includes the original `File` object to pass to the `submitPassword` function once the user clicks the Add password button.

Reply:
It is there to keep the render code cleaner...
- Already uploaded files only include a `name` property, but no `file`
- Files that need a password include both a `file` (to provide for upload once a password is added) and `name` to render a file name without needing the render to check what type of entry it is rendering

Reply:
I am open to suggestions to make this better.